src/io/parsers/MOL2Parser.js

```diff
@@ -0,0 +1,265 @@
+import * as THREE from 'three';
+import Parser from './Parser';
+import chem from '../../chem';
+import MOL2Stream from './MOL2Stream';
+
+const {
+  Complex,
+  Element,
+  Bond,
+  Molecule,
+} = chem;
+
+const orderMap = {
+  un: 0,
+  1: 1,
+  2: 2,
+  3: 3,
+  ar: 1,
+  am: 1,
+  nc: 0,
+  du: 1,
+};
+const typeMap = {
+  un: Bond.BondType.UNKNOWN, // unknown (cannot be determined from the parameter tables)
+  1: Bond.BondType.COVALENT, // single
+  2: Bond.BondType.COVALENT, // double
+  3: Bond.BondType.COVALENT, // triple
+  ar: Bond.BondType.AROMATIC, // aromatic
+  am: Bond.BondType.COVALENT, // amide
+  nc: Bond.BondType.UNKNOWN, // not connected
+  du: Bond.BondType.COVALENT, // dummy
+};
+
+const resNumberRegex = /\d+$/;
+const spacesRegex = /\s+/;
+
+function splitToFields(str) {
+  return str.trim().split(spacesRegex);
+}
+/* There is no jsdoc documentation because of eslint corrections:
+ * not all Parser methods are implemented
+ */
+
+class MOL2Parser extends Parser {
+  constructor(data, options) {
+    super(data, options);
+
+    this._complex = null;
+    this._chain = null;
+    this._residue = null;
+    this._compoundIndx = -1;
+
+    this._molecules = [];
+    this._molecule = null;
+
+    this._serialAtomMap = {};
+
+    this._options.fileType = 'mol2';
+  }
+
+  _parseMolecule(stream) {
+    stream.getHeaderString('MOLECULE');
+
+    const { metadata } = this._complex;
+    metadata.name = stream.getNextString();
+    metadata.format = 'mol2';
+
+    this._molecule = { _index: '', _chains: [] };
+    this._molecule._index = this._compoundIndx + 1;
+    this._molecules.push(this._molecule);
+  }
+
+  /* Atom format description:
+   * atomId atomName x y z element [resSeq [resName [charge [statusBit]]]]
+   * statusBits is the internal SYBYL status bits associated with the atom.
+   * These should never be set by the user.
+   * Source: http://chemyang.ccnu.edu.cn/ccb/server/AIMMS/mol2.pdf
+   */
+  _parseAtoms(stream, atomsNum) {
+    let curStr = stream.getHeaderString('ATOM');
+
+    for (let i = 0; i < atomsNum; i++) {
+      curStr = stream.getNextString();
+      const parsedStr = splitToFields(curStr);
```

Comment:
Please double-check that fields are whitespace-delimited. There's a strange example in the MOL2 format specs (note that `subst_name` and `charge` fields are not separate):

```plain
1      C1    1.207  2.091  0.000  C.ar  1      BENZENE0.000
```

Reply:
This example seemed strange to me. Unfortunately, links to the official tripos site are invalid now. One of them should provide the sample file from the documentation: http://www.tripos.com/mol2/mol2_format3.html
The only way I found to check the real example is Wayback Machine, there is a very old site version in the archive dated by 2006. You could follow the link below and ensure these fields are separated, there seems to be a typo in the spec: http://web.archive.org/web/20060316093219/http://www.tripos.com/mol2/mol2_format3.html
Also I found the statement in the spec, it could be found on the attached a screenshot.
<img width="624" alt="Screenshot 2019-09-20 at 19 29 01" src="https://user-images.githubusercontent.com/25665345/65343137-f880f600-dbdc-11e9-822d-9b2731ae2c57.png">